test/e2e/framework/deployments.go

```diff
@@ -0,0 +1,127 @@
+/*
+SPDX-License-Identifier: Apache-2.0
+
+Copyright Contributors to the Submariner project.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package framework
+
+import (
+	"context"
+
+	"github.com/submariner-io/shipyard/test/e2e/framework"
+
+	appsv1 "k8s.io/api/apps/v1"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+func (f *Framework) NewNetShootDeployment(cluster framework.ClusterIndex) *corev1.PodList {
+	var replicaCount int32 = 1
```

Comment:
The replicaCount is set to 1, so it looks like the requirement is just to spawn a Pod which can be used to run a command. We already have APIs in Shipyard to spawn customPods which can be used for this purpose - https://github.com/submariner-io/shipyard/blob/6b99f918e45e1dcae9e977cb294cf09065266c44/test/e2e/framework/network_pods.go#L124

and its used in Submariner e2e tests here - https://github.com/submariner-io/submariner/blob/c047698ccbf1fcc43b2dcd7c0773acc902914f61/test/e2e/framework/dataplane.go#L143

I think we can remove this API and directly use the CustomPod interface from Shipyard.